internal/db/schema/migrations/oss/postgres/70/02_sessions.up.sql

```diff
@@ -0,0 +1,162 @@
+    'session_target_address entries represent a session that is using a network address that is assigned directly to a Target.';
+
+  create trigger immutable_columns before update on session_target_address
+    for each row execute procedure immutable_columns('session_id');
+
+  create or replace function cancel_session_with_null_target_address_fk() returns trigger
+  as $$
+  begin
+    if new.target_id is null then
+      perform cancel_session(new.session_id);
+      delete from session_target_address where session_id = new.session_id;
+    end if;
+    return new;
+  end;
+  $$ language plpgsql;
+  
+  create trigger cancel_session_with_null_target_address_fk after update of target_id on session_target_address
+    for each row execute procedure cancel_session_with_null_target_address_fk();
+
+  create table session_host_set_host (
+    session_id wt_public_id primary key,
+    host_set_id wt_public_id,
+    host_id wt_public_id,
+    -- the following foreign key constraints for host_set_id & host_id is set to null on a delete
+    -- because we want to be able to invoke a trigger to call cancel_session
+    -- when the references column is deleted
+    -- rather than canceling a session for any reason that may cause a row to be deleted.
+    constraint host_set_fkey foreign key (host_set_id)
+        references host_set (public_id)
+        on delete set null
+        on update cascade,
+    constraint host_fkey foreign key (host_id)
+        references host (public_id)
+        on delete set null
+        on update cascade,
+    constraint session_fkey foreign key (session_id)
+        references session (public_id)
+        on delete cascade
+        on update cascade,
+    constraint session_id_host_set_id_host_id_uq
+        unique(session_id, host_set_id, host_id)
+  );
+  comment on table session_host_set_host is
+    'session_host_set entries represent a session that is using a Host Set.';
+
+  create trigger immutable_columns before update on session_host_set_host
+    for each row execute procedure immutable_columns('session_id');
+
+  create or replace function cancel_session_with_null_host_source_fk() returns trigger
+  as $$
+  begin
+    if new.host_set_id is null or new.host_id is null then
+      perform cancel_session(new.session_id);
+      delete from session_host_set_host where session_id = new.session_id;
+    end if;
+    return new;
+  end;
+  $$ language plpgsql;
+  
+  create trigger cancel_session_with_null_host_source_fk after update of host_set_id, host_id on session_host_set_host
+    for each row execute procedure cancel_session_with_null_host_source_fk();
+
+  drop view session_list;
+  alter table session
+    drop constraint session_host_id_fkey,
+    drop constraint session_host_set_id_fkey,
+    drop column host_id,
+    drop column host_set_id
+  ;
+
+  -- Replaces trigger from 44/04_sessions.up.sql
+  drop trigger insert_session on session;
+  create or replace function insert_session() returns trigger
```

Comment:
this should be `create` not `create or replace`

Reply:
Same issue for this "database migration will return an error for using the same function name twice"

Reply:
`create or replace` is correct